### PR TITLE
Natural Join

### LINKED ISSUE #495: Natural Join
Implement the natural join in the SQLToASTTranslator should be pretty straight-forward.

In case this takes longer than an hour, we should reconsider if we really want this. So far, I can't think of a benchmark that needs it.

### PR DESCRIPTION
**Pull request by mrks**
fixes #495

### CODE REVIEW

**Comment by mrks** [comment]
Could you give an example? Looking at the standard, I don't see how this becomes any more complicated than "find columns with the same name and join them"

File: src/test/sql/sql_to_ast_translator_test.cpp [hunk 1]
```diff
@@ -330,22 +330,30 @@ TEST_F(SQLToASTTranslatorTest, SelectOuterJoin) {
   EXPECT_EQ((*join_node->join_column_ids()).second, ColumnID{0} /* "a" */);
 }
 
-// TODO(mp): Natural Joins are not translated correctly yet. Resolving matching columns is not implemented so far.
-TEST_F(SQLToASTTranslatorTest, DISABLED_SelectNaturalJoin /* #495 */) {
-  const auto query = "SELECT * FROM table_a AS a NATURAL JOIN table_b AS b;";
+TEST_F(SQLToASTTranslatorTest, SelectNaturalJoin) {
+  const auto query = "SELECT * FROM table_a NATURAL JOIN table_c;";
```
Comment (TwoLaid) [inline src/test/sql/sql_to_ast_translator_test.cpp]: For example, switching columns in a subquery currently breaks the natural join:

```
(debug)> load src/test/tables/int_float.tbl A
(debug)> load src/test/tables/int_float5.tbl C

(debug)> SELECT * FROM A NATURAL JOIN C;
=== Columns
|       a|       b|       d|
|     int|   float|   float|
=== Chunk 0 ===
|     123|   456.7|    23.7|
===

(debug)> SELECT * FROM A NATURAL JOIN (SELECT d, a FROM C) AS T2;
Exception thrown while compiling query plan:
  No matching columns for natural join found
```

Comment (mrks) [reply src/test/sql/sql_to_ast_translator_test.cpp]: I'll look into this.

Comment (mrks) [reply src/test/sql/sql_to_ast_translator_test.cpp]: Fixed it. `set_intersection` expected the input to be sorted. Good catch, thank you!

**Comment by mrks** [comment]
There is another issue (#511) in the statistics component that leads to the failing test.

You said "for example" - are you aware of any other issues?

File: src/test/sql/sql_to_ast_translator_test.cpp (hunk 1 above)
Comment (TwoLaid) [reply src/test/sql/sql_to_ast_translator_test.cpp]: I cant think of another example right now, sorry

Comment (mrks) [reply src/test/sql/sql_to_ast_translator_test.cpp]: Don't be sorry - I hope there are none. ;)

**Pull request merged**